### PR TITLE
allow config metadata validator in md.style.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @superyyrrzz @hellosnow @ansyral @qinezh

### CODE REVIEW

File: src/Microsoft.DocAsCode.Dfm/MarkdownValidators/MarkdownMetadataValidationRule.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Dfm.MarkdownValidators
+{
+    using System.ComponentModel;
+
+    using Newtonsoft.Json;
+
+    public class MarkdownMetadataValidationRule
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Dfm/MarkdownValidators/MarkdownMetadataValidationRule.cs]: >MarkdownMetadataValidationRule [](start = 17, length = 30)

seems same to MarkdownValidationRule?

**Comment by ansyral** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs [hunk 2]
```diff
@@ -103,6 +94,34 @@ public void Build(IEnumerable<DocumentBuildParameters> parameters, string output
...
+        private IEnumerable<IInputMetadataValidator> GetMetadataRules(DocumentBuildParameters parameter)
+        {
+            try
+            {
+                var mvb = MarkdownValidatorBuilder.Create(_container, parameter.Files.DefaultBaseDir, parameter.TemplateDir);
+                return mvb.GetEnabledMetadataRules();
+            }
+            catch (Exception)
+            {
+                return Enumerable.Empty<IInputMetadataValidator>();
```
Comment (chenkennt) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: >return [](start = 16, length = 6)

log here?

File: src/Microsoft.DocAsCode.Dfm/MarkdownValidators/MarkdownSytleConfig.cs [hunk 3]
```diff
@@ -9,6 +9,8 @@ public class MarkdownSytleConfig
     {
         public const string MarkdownStyleFileName = "md.style";
 
+        [JsonProperty("metadata")]
```
Comment (chenkennt) [inline src/Microsoft.DocAsCode.Dfm/MarkdownValidators/MarkdownSytleConfig.cs]: >metadata [](start = 23, length = 8)

metadataRules?

File: src/Microsoft.DocAsCode.Dfm/MarkdownValidators/MarkdownMetadataValidationRule.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Dfm/MarkdownValidators/MarkdownMetadataValidationRule.cs]: yes, they are same now.

---
In reply to: [73829630](https://github.com/dotnet/docfx/pull/579#discussion_r73829630) [](ancestors = 73829630)

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by chenkennt** [comment]
:shipit:

**Pull request merged**